### PR TITLE
New docs and tests

### PR DESCRIPTION
**Pull request by Shqrdx**
Docs and tests for

- FillMode
- GraphicsAdapter.DriverType
- GraphicsDeviceStatus
- GraphicsProfile
- IndexElementSize

### CODE REVIEW

File: MonoGame.Framework/Graphics/GraphicsAdapter.cs [hunk 1]
```diff
@@ -19,13 +19,25 @@ namespace Microsoft.Xna.Framework.Graphics
...
+            /// Hardware device been used for rendering. Maximum speed and performance.
+            /// </summary>
             Hardware,
+            /// <summary>
+            /// Emulates the hardware device on CPU. Slowly, only for testing.
+            /// </summary>
             Reference,
+            /// <summary>
+            /// Useful when <see cref="DriverType.Hardware"/> acceleration does not work.
+            /// </summary>
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/GraphicsAdapter.cs]: We should note that `Reference` and `FastSoftware` only work on DirectX platforms at the moment.  Not sure if we're better off listing specific platforms it works on like "Windows desktop and Windows Store".

Don't think those work on Windows Phone... someone needs to check.

File: MonoGame.Framework/Graphics/GraphicsProfile.cs [hunk 2]
```diff
@@ -1,49 +1,18 @@
...
+// file 'LICENSE.txt', which is part of this source code package.
+
 namespace Microsoft.Xna.Framework.Graphics
 {
 	public enum GraphicsProfile
 	{
-        Reach =  0, // 	Use a limited set of graphic features and capabilities, allowing the game to support the widest variety of devices, including all Windows-based computers and Windows Phone.
-        HiDef =  1, // 	Use the largest available set of graphic features and capabilities to target devices, such as an Xbox 360 console and a Windows-based computer, that have more enhanced graphic capabilities.        
+        /// <summary>
+        /// Use a limited set of graphic features and capabilities, allowing the game to support the widest variety of devices, including all Windows-based computers and Windows Phone.
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/GraphicsProfile.cs]: We shouldn't talk about specific platforms here, the "Windows-based computers and Windows Phone" part.  Instead we should say it works on all MonoGame target platforms.

File: MonoGame.Framework/Graphics/GraphicsProfile.cs [hunk 3]
```diff
@@ -1,49 +1,18 @@
...
+        /// </summary>
+        Reach,
+        /// <summary>
+        /// Use the largest available set of graphic features and capabilities to target devices, such as an Xbox 360 console and a Windows-based computer, that have more enhanced graphic capabilities.
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/GraphicsProfile.cs]: Again... we shouldn't mention Xbox or Windows here.

File: MonoGame.Framework/Graphics/States/FillMode.cs [hunk 4]
```diff
@@ -1,8 +1,21 @@
...
+    /// Defines options for filling the primitive.
+    /// </summary>
     public enum FillMode
     {
+        /// <summary>
+        /// Draw solid faces for each primitive.
+        /// </summary>
         Solid,
+        /// <summary>
+        /// Draw lines connecting the vertices that define a primitive face.
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/States/FillMode.cs]: This line is identical to the XNA docs on MSDN.  We cannot copy Microsoft's docs...  this is the same as copying their code... it is a copyright violation.

Comment (Shqrdx) [inline MonoGame.Framework/Graphics/States/FillMode.cs]: Suggest your version.

File: MonoGame.Framework/Graphics/GraphicsAdapter.cs (hunk 1 above)
Comment (Shqrdx) [inline MonoGame.Framework/Graphics/GraphicsAdapter.cs]: The only direct usage of this enum is Adapter.UseDriverType(and that described there), I doubt this is good idea to write it in enum description but if you wants..

File: MonoGame.Framework/Graphics/States/FillMode.cs (hunk 4 above)
Comment (Shqrdx) [inline MonoGame.Framework/Graphics/States/FillMode.cs]: @tomspilman Ok, i changed it myself.

File: MonoGame.Framework/Graphics/GraphicsAdapter.cs (hunk 1 above)
Comment (Shqrdx) [inline MonoGame.Framework/Graphics/GraphicsAdapter.cs]: I changed DriverType description itself - added "Usable only on DirectX platforms for now." than all these members are meaning to support only on DirectX.

**Comment by Shqrdx** [comment]
@tomspilman How it is now ?

**Comment by tomspilman** [comment]
Good enough... merging!

**Pull request merged**